test/TritonGPU/loop-pipeline-indirect-load.mlir

```diff
@@ -0,0 +1,90 @@
+// RUN: triton-opt %s -split-input-file -tritongpu-pipeline=num-stages=2 | FileCheck %s
+// CHECK-LABEL: @indirect_load_two_stages
+// CHECK: scf.for
+// CHECK: tt.dot
+// CHECK: tt.load
+// CHECK: async_copy_global_to_local
+// CHECK: async_copy_global_to_local
```

Comment:
could we add a `tritongpu-pipeline=num-stages=2` run in loop-pipeline instead and re-use the indirect load test there for checks?

Reply:
Tried "triton-opt test/TritonGPU/loop-pipeline.mlir -split-input-file -tritongpu-pipeline=num-stages=2 -canonicalize", it didn't expose this issue of "operation scheduled before its operands".